### ISSUE TITLE
Calibration?

### ISSUE THREAD

**Issue by Pshemas** [tb2097/wacom-gui#28]
Super happy to find this one - thank you for the work on the app. Can't wait to check it on OpenSUSE Leap 15 with Xfce.
Does it have any calibration capabilities? Would be great for fine tuning Cintiqs.

**Comment by tb2097** [comment]
By calibration, do you mean for setting the active area of the display?  I haven't implemented "touch" calibration like it does in Windows but it does use the information provided by the Cintiq to determine it's full screen resolution/touch area.  I haven't had any issues with the Cintiq devices I have calibrated so far, although I do plan to hopefully implement a similar feature.  Pressure curve etc for the pen are possible. I haven't done anything yet with different pen types, but all I have access to right now is a standard pen so testing would be impossible.

**Comment by Pshemas** [comment]
no, I mean something like this:
https://youtu.be/mXlatUbfyAk?t=72
-crucial for Cintiqs and other display tablets, allows you to minimize parallax in your setup. Gnome's Wacom panel has it, also I've seen separate linux utility that did similar thing (can't recall the name, if I recall correctly originally it was used for setting up touch displays - my wild guess is that Gnome based their implementation on that).

**Comment by tb2097** [comment]
Technically, that is what the screen mapping is doing.  However, I don't have options configured for doing "partial" mapping, which is what you are looking for.  My initial plan was to use [[https://github.com/tias/xinput_calibrator]], however it has an issue with multiple screens where it will display across multiple displays, even if you just want to calibrate the cintiq.  I started looking into the code to see if I can reverse engineer what they are doing but I don't have it in a working state yet.

I'd suggest trying to set it to the Cintiq's display output and see how close it is, for now at least.  I'm a little busy at work for the next bit so I don't think I'll have time to further test the code for a while.

**Comment by Pshemas** [comment]
yeah, but functionally this is different thing. One maps the tablet to correct screen, the other fine tunes the mapping offsets so the paralax on display tablets is less noticeable (also takes into account any inconsistencies between display/tablet made during manufacturing).

No pressure here - I'm already super grateful that you did something that has been lacking on the Linux side of things for quite a while (guiding creatives to figure out command combos in the command line is not working :D ). I'm just saying this extra functionality will take it up a notch - and in fact I think your solution will be more complete than the one found in Gnome's panel (which is almost perfect - it lacks option to set up scrollbars / scrollwheels and I've envoutered some limitations in multi tablet configurations).

**Comment by tb2097** [comment]
I'm glad you like what I have so far :)  The "calibration" option, since the code is effectively the same on the backend for setting up a targeted area of the screen, is the next feature I want to address.  I just need to understand how it is being done in the background ie. upper left is drawn at 50, 50, but the pen detects as touching at 48, 51.  How is the difference rectified?  I just need time to examine the code from xinput_calibrator to know what it is doing.  It will be coming, eventually.
